src/setup/demo.md

```diff
@@ -0,0 +1,236 @@
+
+The `add_dataset_objective.py` script is in charge of registering data to the `organisation`.
+
+Before running the script, please make sure you have updated:
+
+- `profile` in `client = substra.Client(profile_name="<PROFILE>")`, using the profile you created before
+- `DATASET`: `name` & `permissions` (demo organisations are named `MyOrg1MSP` & `MyOrg2MSP`)
+- `OBJECTIVE`: `name` & `permissions` (demo organisations are named `MyOrg1MSP` & `MyOrg2MSP`)
+- any different path in regard to project file structure
+
+You can then run the command:
+
+```sh
+python3 add_dataset_objective.py
+```
+
+### Add your algorithm
+
+You will need to update the `algo.py` file as well as the `description.md` & `Dockerfile`, and finally push it to the platform with the dedicated script:
+
+```sh
+python3 add_train_algo.py
+```
+
+In the `algo.py` file, please make sure you have updated:
+
+- `profile` in `client = substra.Client(profile_name="<PROFILE>")`, using the profile you created before
+- `ALGO`: `name` &  `permissions` (demo organisations are named `MyOrg1MSP` & `MyOrg2MSP`)
+
+If you need to test that your algorithm is working without having access to the data, you can use the CLI method `substra run-local` that will use the `fake_data` method from the `opener.py`:
+
+```sh
+substra run-local assets/algo \
+  --train-opener=assets/dataset/opener.py \
+  --test-opener=assets/dataset/opener.py \
+  --metrics=assets/objective/ \
+  --fake-data-samples
+```
+
+### Training & Testing tasks
+
+```sh
+# Train
+python3 assets/algo/algo.py train \
+  --debug \
+  --opener-path assets/dataset/opener.py \
+  --data-samples-path assets/train_data_samples \
+  --output-model-path assets/model/model \
+  --log-path assets/logs/train.log
+
+# Predict
+python assets/algo/algo.py predict \
+  --debug \
+  --opener-path assets/dataset/opener.py \
+  --data-samples-path assets/train_data_samples \
+  --output-predictions-path assets/pred-train.csv \
+  --models-path assets/model/ \
+  --log-path assets/logs/train_predict.log \
+  model
+
+# Performance
+python assets/objective/metrics.py \
+  --debug \
+  --opener-path assets/dataset/opener.py \
+  --data-samples-path assets/train_data_samples \
+  --input-predictions-path assets/pred-train.csv \
+  --output-perf-path assets/perf-train.json \
+  --log-path assets/logs/train_metrics.log
+
+# Test / predict
+python assets/algo/algo.py predict \
+  --debug \
+  --opener-path assets/dataset/opener.py \
+  --data-samples-path assets/test_data_samples \
+  --output-predictions-path assets/pred-test.csv \
+  --models-path assets/model/ \
+  --log-path assets/logs/test_predict.log \
+  model
+
+# Test / performance
+python assets/objective/metrics.py \
+  --debug \
+  --opener-path assets/dataset/opener.py \
+  --data-samples-path assets/test_data_samples \
+  --input-predictions-path assets/pred-test.csv \
+  --output-perf-path assets/perf-test.json \
+  --log-path assets/logs/test_metrics.log
+```
```

Comment:
Ok - so I guess from this that:

- we start with training a model
- then we launch a predict task, which seems to be needed to be able to launch the computation of the performance metric - is that it?
- on the test set, same thing: predict + performance.

If this is correct, could you please group this in 2 groups: 1 for train, 1 for test? It will be clearer.

Reply:
Okay!